src/helpers/card-deck.helper.ts

```diff
@@ -1,60 +1,80 @@
-import { CardType } from 'src/models/card-type.model';
-
-export class CardDeckHelperService {
-  public possibleSuits: string[] = ['spades', 'diams', 'clubs', 'hearts'];
-  public possiblevalues: string[] = [
-    'A',
-    '2',
-    '3',
-    '4',
-    '5',
-    '6',
-    '7',
-    '8',
-    '9',
-    '10',
-    'J',
-    'Q',
-    'K',
-  ];
-  public createdCardDeck: Array<CardType> = [];
-
-  public construcDesck(): Array<CardType> {
-    let deck = new Array<CardType>();
-
-    for (let i = 0; i < this.possibleSuits.length; i++) {
-      for (let x = 0; x < this.possiblevalues.length; x++) {
-        let card: CardType = {
-          value: this.possiblevalues[x],
-          suit: this.possibleSuits[i],
-          icon: `&${this.possibleSuits[i]};`,
-          Sips: 0,
-          SipsSelected: false,
-        };
-        deck.push(card);
-        this.createdCardDeck.push(card);
-      }
-    }
-
-    return deck;
-  }
-
-  public getRandomCard() {
-    const card =
-      this.createdCardDeck[
-        Math.floor(Math.random() * this.createdCardDeck.length)
-      ];
-
-    const toDelete = this.createdCardDeck.indexOf(card, 0);
-
-    if (toDelete > -1) {
-      this.createdCardDeck.splice(toDelete, 1);
-    }
-
-    return card;
-  }
-
-  public resetCards(): void {
-    this.createdCardDeck = [];
-  }
-}
+import {CardType} from 'src/models/card-type.model';
+import {LocalService} from "../app/services/local/local.service";
+import {Injectable} from "@angular/core";
+
+@Injectable({
+  providedIn: 'root',
+})
+
+export class CardDeckHelperService {
+
+  public possibleSuits: string[] = ['spades', 'diams', 'clubs', 'hearts'];
+  public possiblevalues: string[] = [
+    'A',
+    '2',
+    '3',
+    '4',
+    '5',
+    '6',
+    '7',
+    '8',
+    '9',
+    '10',
+    'J',
+    'Q',
+    'K',
+  ];
+  public createdCardDeck: Array<CardType> = [];
+
+
+  constructor(
+    public localSrv: LocalService
+  ) {
+    const sessionCardDeck = JSON.parse(this.localSrv.getData('cardDeck') as string);
+    this.createdCardDeck = sessionCardDeck ? sessionCardDeck : [];
+  }
+
+  public construcDesck(): Array<CardType> {
+    this.resetCards();
+    let deck = new Array<CardType>();
+
+    for (let i = 0; i < this.possibleSuits.length; i++) {
+      for (let x = 0; x < this.possiblevalues.length; x++) {
+        let card: CardType = {
+          value: this.possiblevalues[x],
+          suit: this.possibleSuits[i],
+          icon: `&${this.possibleSuits[i]};`,
+          swallow: 0,
+          selected: false,
+        };
+        deck.push(card);
+        this.createdCardDeck.push(card);
+      }

```

Comment:
```suggestion
      card: CardType = {
        suit: this.possibleSuits[i],
        icon: `&${this.possibleSuits[i]};`,
        swallow: 0,
        selected: false,
      };

      for (let x = 0; x < this.possiblevalues.length; x++) {
        card.value = this.possiblevalues[x];
        this.createdCardDeck.push(card);
      }
```

Reply:
Only king in the deck 
 card: CardType = {
        suit: this.possibleSuits[i],
        icon: `&${this.possibleSuits[i]};`,
        swallow: 0,
        selected: false,
      };
      
      Must be in second for
      
      Fix in my code